lib/basedriver/capabilities.js

```diff
@@ -152,16 +170,25 @@ function parseCaps (caps, constraints = {}, shouldValidateCaps = true) {
     } catch (ign) { }
   }
 
-  // If there was no match, use the de-facto requiredCaps on its own
-  matchedCaps = matchedCaps || requiredCaps;
-
   // Returns variables for testing purposes
-  return {requiredCaps, allFirstMatchCaps, validatedFirstMatchCaps, matchedCaps};
+  return {requiredCaps, allFirstMatchCaps, validatedFirstMatchCaps, matchedCaps, errors};
 }
 
 // Calls parseCaps and just returns the matchedCaps variable
 function processCapabilities (caps, constraints = {}, shouldValidateCaps = true) {
-  return parseCaps(caps, constraints, shouldValidateCaps).matchedCaps;
+  const {matchedCaps, errors} = parseCaps(caps, constraints, shouldValidateCaps);
+
+  if (matchedCaps === null) {
+    if (_.isArray(caps.firstMatch) && caps.firstMatch.length > 1) {
+      // If there was more than one 'firstMatch' cap, indicate that we couldn't find a matching capabilities set and show all the errors
+      throw new Error(`Could not find matching capabilities from ${JSON.stringify(caps)}:\n ${errors.join('\n')}`);
+    } else {
+      // Otherwise, just show the singular error message
+      throw new Error(errors[0]);
```

Comment:
should we check if errors is not empty first?

Reply:
It can't be empty. If it found an error in a singleton `firstMatch` array then there was exactly one error.